### PR TITLE
Add GPU RNNT Loss

### PR DESCRIPTION
**Pull request by carolineechen**
Summary: In #1479, we added support for CPU RNNT loss. This PR adds a GPU version of RNNT loss

Differential Revision: D28128853

### CODE REVIEW

File: build_tools/setup_helpers/extension.py [hunk 1]
```diff
@@ -38,6 +38,7 @@ def _get_build(var, default=False):
 _BUILD_KALDI = False if platform.system() == 'Windows' else _get_build("BUILD_KALDI", True)
 _BUILD_TRANSDUCER = _get_build("BUILD_TRANSDUCER")
 _USE_ROCM = _get_build("USE_ROCM")
+_WITH_CUDA = True if torch.cuda.is_available() else False
```
Comment (mthrok) [inline build_tools/setup_helpers/extension.py:41]: PyTorch core uses `USE_CUDA`. https://github.com/pytorch/pytorch/search?l=CMake&q=USE_CUDA 
So we should align with them.

**Comment by carolineechen** [comment]
@malfet @seemethere Could I get your input on supporting custom CUDA kernel in torchaudio?

One of our features (RNN transducer loss) for our the upcoming release involves a custom CUDA kernel, but this isn't something that torchaudio currently supports. I have a minimal working CUDA build in this PR that compiles successfully on AWS and in the CircleCI builds (mac builds are failing on master for an unrelated reason), but I’m not very familiar with CUDA builds. Are there any additional flags that we should include (torchvision includes [these](https://github.com/pytorch/vision/blob/19ad0bbc5e26504a501b9be3f0345381d6ba1efc/CMakeLists.txt#L8)), or anything platform specific (ex/ for Windows) that we should add?

File: build_tools/setup_helpers/extension.py [hunk 2]
```diff
@@ -38,6 +38,7 @@ def _get_build(var, default=False):
 _BUILD_KALDI = False if platform.system() == 'Windows' else _get_build("BUILD_KALDI", True)
 _BUILD_TRANSDUCER = _get_build("BUILD_TRANSDUCER")
 _USE_ROCM = _get_build("USE_ROCM")
+_USE_CUDA = True if torch.cuda.is_available() else False
```
Comment (malfet) [inline build_tools/setup_helpers/extension.py:41]: Nit
```suggestion
_USE_CUDA = torch.cuda.is_available()
```

File: torchaudio/csrc/rnnt/gpu/compute_alphas.cu [hunk 3]
```diff
@@ -0,0 +1,73 @@
+#include <THC/THC.h>
```
Comment (malfet) [inline torchaudio/csrc/rnnt/gpu/compute_alphas.cu:1]: Do you really need to include THC?

**Review by malfet — APPROVED** [review]

File: torchaudio/csrc/rnnt/gpu/compute_alphas.cu (hunk 3 above)
Comment (carolineechen) [reply torchaudio/csrc/rnnt/gpu/compute_alphas.cu:1]: I can replace THC with something like `<c10/cuda/CUDAStream.h>` to get the same result -- is there a standard for supporting the `getCurrentCUDAStream` operator?

File: test/torchaudio_unittest/rnnt/rnnt_loss_cuda_test.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import torch
+from .rnnt_loss_impl import RNNTLossTest
+from torchaudio_unittest import common_utils
+from .utils import skipIfNoTransducer
+
+
+@skipIfNoTransducer
```
Comment (vincentqb) [inline test/torchaudio_unittest/rnnt/rnnt_loss_cuda_test.py:7]: nit: just realized the name has not been standardized from Transducer to RNNT, but this can be changed after this PR

**Review by vincentqb — APPROVED** [review]
Awesome, LGTM!

File: torchaudio/csrc/rnnt/gpu/compute_alphas.cu (hunk 3 above)
Comment (cpuhrsch) [reply torchaudio/csrc/rnnt/gpu/compute_alphas.cu:1]: Including <c10/cuda/CUDAStream.h> is fine. THC is legacy and on its way out.
